builtin/providers/azure/provider_test.go

```diff
@@ -11,6 +11,18 @@ import (
 var testAccProviders map[string]terraform.ResourceProvider
 var testAccProvider *schema.Provider
 
+const testAccSecurityGroupName = "terraform-security-group"
+
+// testAccStorageServiceName is used as the name for the Storage Service
+// created in all storage-related tests.
+// It is much more convenient to provide a Storage Service which
+// has been created beforehand as the creation of one takes a lot
+// and would greatly impede the multitude of tests which rely on one.
+// NOTE: the storage container should be located in `West US`.
+var testAccStorageServiceName = os.Getenv("AZURE_STORAGE")
```

Comment:
The constants are a bit dodgy; yes, but I placed them there for a couple of reasons:
* it may seem superficial, but they aid the changing of these values in the rare cases where one might like to run the tests but somehow already has a `terraform-security-group` security group up there (it actually happened to me).
* they greatly aid practicality in tests of 'sub-resources' that all stem from the same super-resource and thus there's no point in fetching that particular value for each when it's always the same for all of them (the storage elements example).